include/sampleheap.hpp

```diff
@@ -54,11 +61,11 @@ class SampleHeap : public SuperHeap {
         _freedLastMallocTrigger(false) {
       
       signal_init_lock.lock();
-      sighandler_t old_malloc = signal(MallocSignal, SIG_IGN);
+      SIGHANDLER_TYPE old_malloc = signal(MallocSignal, SIG_IGN);
```

Comment:
same comment as above, just use `auto` (or `decltype` if you really need a type declaration)